rotkehlchen/tests/unit/test_protocol_balances.py

```diff
@@ -182,3 +184,31 @@ def test_velodrome_v2_staking_balances(
         amount=FVal('0.043087772070655563'),  # staked in gauge
         usd_value=FVal('0.0646316581059833445'),
     )
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [['0xB9a398aAA3B8890aBbd49e2161Aff629A550152e']])
+def test_thegraph_balances(
+        ethereum_inquirer: 'EthereumInquirer',
+        ethereum_transaction_decoder: 'EthereumTransactionDecoder',
+        ethereum_accounts: list[ChecksumEvmAddress],
+        inquirer: 'Inquirer',  # pylint: disable=unused-argument
+) -> None:
+    tx_hex = deserialize_evm_tx_hash('0x009cd8eccb0637a381d00082056654c534c9e74aa7c1b60196c50b6638236c08')  # noqa: E501
+    get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=ethereum_transaction_decoder.database,
+        tx_hash=tx_hex,
+    )
+    thegraph_balances_inquirer = ThegraphBalances(
+        database=ethereum_transaction_decoder.database,
+        evm_inquirer=ethereum_inquirer,
+        chain_id=ChainID.ETHEREUM,
+    )
+    thegraph_balances = thegraph_balances_inquirer.query_balances()
+    user_balance = thegraph_balances[ethereum_accounts[0]]
+    token = A_GRT.resolve_to_evm_token()
+    assert user_balance[token] == Balance(
+        amount=FVal('1293.499999999999900000'),
+        usd_value=FVal('1940.24999999999985'),
+    )
```

Comment:
I believe is enough doing

```suggestion
    assert user_balance[A_GRT] == Balance(
        amount=FVal('1293.499999999999900000'),
        usd_value=FVal('1940.24999999999985'),
    )
```

Reply:
Expects `EvmToken` as key:
```bash
error: Invalid index type "Asset" for "dict[EvmToken, Balance]"; expected type "EvmToken"  [index]
```

Reply:
just resolve it in there instead of creating a variable